db/c.cc

```diff
@@ -5632,6 +5673,33 @@ void rocksdb_options_set_min_level_to_compress(rocksdb_options_t* opt,
   }
 }
 
+rocksdb_livefiles_t* rocksdb_livefiles_create() {
+  return new rocksdb_livefiles_t;
+}
+
+void rocksdb_livefiles_add(rocksdb_livefiles_t* lf,
+                           const char* column_family_name, const char* name,
+                           const char* directory, size_t size, int level,
+                           const char* start_key, size_t start_key_len,
+                           const char* end_key, size_t end_key_len,
+                           uint64_t smallest_seqno, uint64_t largest_seqno,
+                           uint64_t num_entries, uint64_t num_deletions) {
```

Comment:
What's a better way to do this that is still easy enough to use from Rust? I could define a struct for it; my original approach was to wrap the underlying struct but the Rust create has its own LiveFile which isn't directly related to the C struct, and I wanted to reuse that as much as possible.

Reply:
What's the purpose of this function if you want to use `LiveFile` from rust-rocksdb instead? Maybe things become a bit clearer for me when I see the `rust-rocksdb` related changes.

Reply:
There is not magic solution here unfortunately. Your other proposal (passing `rocksdb_livefile_t*` file) is what I'd consider a common way to do business across C FFI boundaries. That said, it's not worth optimizing for unless the rocksdb team pushed back on this if you decided to upstream those APIs.

The downside here is if a new field was added to this struct by rocksdb, it won't be copied over from the original input (the export) and it'll end up getting garbage value which can be dangerous.

If we don't need to expose the ability to add livefiles, then I would remove this and only do it when we hit the use case for it.

Reply:
This signature is going to be a maintenance burden over time. I'm going to add a wrapper struct `rocksdb_livefile_t` and expose individual setters over FFI for it.

If I change the `set_files` operation to have "move semantics" we won't need to worry about cloning the input; just pass the wrapped `LiveFile` instance(s) over to the consumer, which will have been appropriately initialized by the default ctor.

I'm going to start a conversation with upstream in parallel; I was hoping not to touch the pre-existing `DB::live_files` implementation too much but maybe it's needed for a more cohesive overall solution.